prereise/gather/griddata/hifld/data_process/generators.py

```diff
@@ -0,0 +1,136 @@
+import pandas as pd
+from powersimdata.utility.distance import haversine
+
+from prereise.gather.griddata.hifld import const
+from prereise.gather.griddata.hifld.data_access import load
+
+
+def floatify(value, default=pd.NA):
+    """Return a float if possible, otherwise return a default value.
+
+    :param object value: value to be coerced to float, if possible.
+    :param object default: value to be returned if float conversion isn't possible.
```

Comment:
Should we use `np.NaN` instead. Its type is a `float` while `pd.NA` is not:
```
>>> type(pd.NA)
<class 'pandas._libs.missing.NAType'>
>>> type(np.NaN)
<class 'float'>
```

Reply:
Good idea. We could go even simpler with just `float("nan")`:
```python
>>> type(float("nan"))
<class 'float'>
```

Reply:
Done.